### ISSUE TITLE
Error Balloon Crash gvim

### ISSUE THREAD

**Issue by lebster** [scrooloose/syntastic#1515]
When I hover my house over an error the balloon shows up fine, but if I click when it is displayed GVIM crashes.

I am on windows 8.1 pro
VIM - Vi IMproved 7.4
Syntastic version: 3.6.0-146 (Vim 704, Windows)

Below are a couple of images. One is to show the error balloons that I am talking about. The second image is what visual studio debugger shows ( not sure if this is useful )

![image](https://cloud.githubusercontent.com/assets/914433/9214645/3d0f7df0-4055-11e5-8f8b-7889fa025496.png)

![image](https://cloud.githubusercontent.com/assets/914433/9214647/425447b4-4055-11e5-98dd-2aa971b833ce.png)

**Comment by lcd047** [comment]
This has nothing to do with syntastic.  It seems to be a bug in Vim, please report it to vim_dev.

**Comment by chrisbra** [comment]
How is this reproducible?

**Comment by lebster** [comment]
First put the following in your vimrc.

execute pathogen#infect()
syntax enable


Then create a file named blah.js with the text
blah +++; 


save then hover your mouse over the error and click.

**Comment by lebster** [comment]
@chrisbra i am not sure why but in vim_dev on google my msgs keep getting deleted. Here is the msg I was trying to send.

I can reproduce the error without any plugins now.

1. I run the following command in cmd.

gvim -u NONE -N -c:"set beval bexpr='foo'"

2. Insert some text then hit esc.
3. Hover mouse over the text.
4. Once foo bubble shows up I click
5. Then it crashes

**Comment by lcd047** [comment]
@chrisbra I can't reproduce the crash on FreeBSD, OpenBSD, and Linux, but there seems to be another problem that might be related: in the same scenario as above I sometimes get a ghost cursor, the old position of the cursor stays highlighted after clicking elsewhere.  It only happens once in a while, always at the first click (it never happens afterwards).  This seems to point to an unitialized variable, or something similar.  For what it's worth.

**Comment by chrisbra** [comment]
Yeah, the Windows code is much different than the other ports. It seems however that we found the crash and a possible fix.
I'll see, if I can find the problem with the ghost cursor as well. This only happens for the first click ever in a gvim Window and than one would have to restart gvim to reproduce, right?

**Comment by lcd047** [comment]
Pretty much, yes.  The ghost cursor is not really a problem in itself since it happens only once in a while and it's short lived, but it might point to a deeper bug. _shrug_

**Comment by killerswan** [comment]
Note, in the conversation [here](https://groups.yahoo.com/neo/groups/vimdev/conversations/topics/82207) a patch for this Windows use after free issue is discussed:
```
diff -r 3efc03b7f01b src/gui_w32.c
--- a/src/gui_w32.c Mon Aug 10 01:38:00 2015 -0400
+++ b/src/gui_w32.c Thu Aug 13 18:28:14 2015 -0400
@@ -4837,7 +4837,8 @@
delete_tooltip(beval)
BalloonEval *beval;
{
- DestroyWindow(beval->balloon);
+ PostMessage(beval->balloon, WM_DESTROY, 0, 0);
+ PostMessage(beval->balloon, WM_NCDESTROY, 0, 0);
}
```

Anyways that's why it was showing up with stack traces pointing to stuff like `comctl32.dll!CToolTipsMgr::GetToolAtPoint()`.  

Was, or is.  I'm not sure what release of vim has or will have the patch.

**Comment by lebster** [comment]
this was included in vim patch 7.4.1208

**Comment by killerswan** [comment]
So it looks like the versions [here](ftp://ftp.vim.org/pub/vim/pc/gvim74.exe) and [here](ftp://ftp.vim.org/pub/vim/pc/gvim74-1024.exe) linked on the Vim.org site won't have it, but the latest from tuxproject.de [here](https://tuxproject.de/projects/vim/) will?

**Comment by chrisbra** [comment]
This has been included as of Vim patch [7.4.829](https://github.com/vim/vim/releases/tag/v7.4.829)
